application/src/main/webapp/WEB-INF/views/main.ftl

```diff
@@ -90,7 +96,19 @@
         </div>
     </li>
 </script>
+
+<script type="text/template" class="recommendationRowTemplate">
+    <li id="<%= data.id %>" title="<%= data.name %>">
+        <div class="input-group">
+            <a type="button" class="btn btn-default btn-block" data-value="<%= data.name %>">
+                <%= data.name %>
+            </a>
+        </div>
+    </li>
+</script>
+
```

Comment:
Actually the recommendationRowTemplate might be extended in future, so I do not see the necessity for a common template for them. :)